llvm/cmake/modules/AddLLVM.cmake

```diff
@@ -1454,7 +1454,12 @@ function(add_unittest test_suite test_name)
     list(APPEND LLVM_COMPILE_FLAGS "-Wno-gnu-zero-variadic-macro-arguments")
   endif()
 
-  set(LLVM_REQUIRES_RTTI OFF)
+  # Enabling of exception handling for unittests means we must enable RTTI.
+  if(NOT (LLVM_REQUIRES_EH OR LLVM_ENABLE_EH))
+    set(LLVM_REQUIRES_RTTI OFF)
```

Comment:
I don't think it's the right approach. Why enabling exceptions support for DPC++ runtime unittests should require enabling exceptions support for LLVM libraries, which are not used by the DPC++ runtime?

Reply:
I missed something. If I placed enabling exceptions to the add_unittest() function, does it affect to the another LLVM libraries?

Reply:
@bader -- I think the issue is that `add_sycl_unittest()` sets `LLVM_REQUIRES_EH` to `1`. It then calls `add_unittest()` which sets `LLVM_REQUIRES_RTTI` to `0` unconditionally. Eventually, we end up in `llvm_update_compile_flags()`, which sees that EH is set to 1 but RTTI is not enabled, and it sets `LLVM_REQUIRES_RTTI` to 1 and we get a warning about that.

This code here is basically saying "if exception handling is not required, don't enable RTTI; if exception handling is required, enable RTTI to avoid later diagnostics about needing to enable it."

I suppose that could be an indication that we should enable RTTI within `add_sycl_unittest()` instead though, so the logic within `add_unittest()` can only be about disabling RTTI if it wasn't already enabled. Is that what you meant by this not being the right approach, or did you have something else in mind?

Reply:
Right. I would expect that fix for SYCL project warnings would be isolated in SYCL project cmake files.
I don't mind enhance LLVM project framework as well, but I'm not sure it's required for our case and if so, it should go to llorg first.

Reply:
I don't think we can make the fix without touching community code, and I think the plan would be for us to come up with a complete fix here, make sure it works for us, and then propose the upstream bits to community. The issue is: if we set `LLVM_REQUIRES_RTTI` to `1` in `add_sycl_unittest()`, it will be reset to `0` in `add_unittest()` and we'll still get the warning from `llvm_update_compile_flags()`. So something needs to change on the LLVM side of things.

I'm not enough of a CMake expert to know the right way to fix this though.

Reply:
@bader What do you suggest?

Reply:
I'm not sure that using LLVM's CMake functions is a good idea for SYCL runtime library unit tests. Instead of using `add_unittest()` from `llvm/cmake/modules/AddLLVM.cmake`, we could have our own CMake function handling unit tests creation which has no LLVM project restrictions like disabled RTTI.
Is there a specific reason for using LLVM's CMake functions?

Reply:
> Is there a specific reason for using LLVM's CMake functions?

Code reuse is my primary reason to want to keep using it. Unit tests through gtest seem like they should all pipe through some common bottleneck rather than be spread to each subproject, IMO.

Reply:
I don't have strong opinion about this particular case. I see both pros and cons of re-using LLVM's CMake files for the runtime library unit tests.